orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -292,9 +292,11 @@ public boolean hasDisableLock(Moniker clusterMoniker, String account, Location l
     if (application == null || !application.details().containsKey("trafficGuards")) {
       return false;
     }
-    List<Map<String, String>> trafficGuards = (List<Map<String, String>>) application.details().get("trafficGuards");
-    List<ClusterMatchRule> rules = trafficGuards.stream().map(guard ->
-      new ClusterMatchRule(guard.get("account"), guard.get("location"), guard.get("stack"), guard.get("detail"), 1)
+    List<Map<String, Object>> trafficGuards = (List<Map<String, Object>>) application.details().get("trafficGuards");
+    List<ClusterMatchRule> rules = trafficGuards.stream()
+      .filter(guard -> !(boolean) guard.getOrDefault("disabled", false))
```

Comment:
The only question that stands out is whether it should be `disabled` and false by default or `enabled` and true by default.

I'm not sure if we have precedent anywhere in Spinnaker? 
(my initial inclination would be to have it ticked in the UI and enabled by default -- I normally equate something ticked to being enabled)

Reply:
We are not consistent in Spinnaker. When we've added flags like this post-hoc, I think we've typically chosen the name and meaning based on what the default behavior should be, to avoid a data migration. With traffic guards, we have external systems (e.g. newt) that generate them on application setup, so there would need to be some coordination if we went with `disabled`.

An alternative would be to use `enabled` and consider the absence of the field to be `true`, but that seems slightly more surprising.

Reply:
It wouldn't be that difficult to set suitable defaults in Front50 for new application-level flags?

Reply:
For pipelines, we use a `disabled` flag; same with application data sources (UI only flag). Not sure about other flags...

Reply:
Triggers, execution window jigger use `enabled`

Reply:
I would like to see this be `enabled` and `true` by default. Let's avoid inverted flags.

Reply:
OMG FINE IT'S `enabled` now